sonic_platform_base/watchdog_base.py

```diff
@@ -0,0 +1,50 @@
+#
+# watchdog_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a hardware watchdog module in SONiC
+#
+
+from . import device_base
+
+
+class WatchdogBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a hardware watchdog module
+    """
+
+    # Possible fan directions
+    FAN_DIRECTION_INTAKE = "intake"
+    FAN_DIRECTION_EXHAUST = "exhaust"
+
+    def arm(self, seconds):
+        """
+        Arm the hardware watchdog with a timeout of <seconds> seconds
```

Comment:
can you describe the behavior more specifically? if the watchdog is already armed and call this api will refresh the watchdog timer.

Reply:
hardware watchdog may have maximum value of the timeout. I think we need to have a error code to return when seconds exceed the maximum value that is support by the hardware.

Reply:
Currently the function will return False if it is unable to arm the watchdog. Would you prefer an integral return value to specify the reason?

Alternatively (or in addition to), we could ask the vendor to define a maximum value which the user can query. Thoughts?

Reply:
Return value is now an integer.